packages/lesswrong/server/ckEditor/ckEditorWebhook.ts

```diff
@@ -101,8 +109,16 @@ async function handleCkEditorWebhook(message: any) {
       const documentUpdatedPayload = payload as CkEditorDocumentUpdated;
       const ckEditorDocumentId = documentUpdatedPayload?.document?.id;
       const postId = ckEditorDocumentIdToPostId(ckEditorDocumentId);
-      const documentContents = await fetchCkEditorCloudStorageDocument(ckEditorDocumentId);
-      await saveOrUpdateDocumentRevision(postId, documentContents);
+      const [documentContents, post] = await Promise.all([
+        fetchCkEditorCloudStorageDocument(ckEditorDocumentId),
+        Posts.findOne(postId)
+      ]);
+
+      if (!post) {
+        throw new Error(`Post with id ${postId} from ckEditor webhook event ${event} not found when trying to save a new revision`);
+      }
+
+      await saveDocumentRevision(post.userId, postId, documentContents);
```

Comment:
Walk me through the change from `saveOrUpdateDocumentRevision` to `saveDocumentRevision`. Given the large number of ckeditor autosaves, I sure do see the logic of a `saveOrUpdateDocumentRevision` function.

Reply:
Also now `saveOrUpdateDocumentRevision` is essentially dead code. I'd remove it if you decide not to use it here.